### ISSUE TITLE
Issues with new stylesheet

### ISSUE THREAD

**Issue by BreakPoints** [dracula/qbittorrent#13]
I appreciate the theme being updated to better support qBittorent v4.5 but I've spotted a number of regressions/oversights.
Edit: Forgot to mention I'm on Windows 11, Qt5

- Buttons no longer have a background so are not clear that they are buttons, for example see the OK button in the update dialog in the screenshot.
- Backgrounds for tabs, such as in the about dialog, look out of place.
- Links, as seen in the about dialog, are too saturated to be easily readable.
- Mouse over highlighting of text darkens enough to become harder to read.
- Personal opinion: the new scroll bars are rather ugly and do not indicate that they are scroll bars (missing up down arrows as they were before).
- Personal opinion: The square download bars are also rather ugly and don't fit the rounded icon design in the rest of the UI.

![Screenshot 2022-11-28 165900](https://user-images.githubusercontent.com/11966090/204337504-32ab8812-71ee-410c-b2ad-d802bf957ad3.png)

**Comment by MahdiMirzade** [comment]
Thanks for the suggestions, Yes, I also planned to rewrite the stylesheet ([on the repo's TODO list](https://github.com/mahdimirzade/qbittorrent/#todo)), but I want an implementation which can follow a certain template and support more themes (onedark, nord, catpuccin, etc) in the future using a simple script.

> Personal opinion: the new scroll bars are rather ugly and do not indicate that they are scroll bars (missing up down arrows as they were before).

I couldn't figure out how to implement the arrows in the stylesheet for the new version, since I used custom icons in the old styles, I used those icons.
But qBittorrent has had a complete new design for their project which I like and removed all of our own icons, but didn't know why the arrow keys showed up white, so I just hid it for the time being. I'll also try to fix this.

> Personal opinion: The square download bars are also rather ugly and don't fit the rounded icon design in the rest of the UI.

Can you explain this more? I don't quite understand the term "square download bars". Are there other designs for download bars? can you link those for me? 

You're right on all of your points and I'll look more into this when I have more time on my hand.

Other styling suggestions are also welcome in this issue for the future.

Best Regards, Mahdi

**Comment by BreakPoints** [comment]
Thanks for hearing me out, I was quite quick on the draw for criticisms when the update was rolled out.

> Can you explain this more? I don't quite understand the term "square download bars". Are there other designs for download bars? can you link those for me?

It is a rather subtle effect, the corner pixels are removed and a slight gradient applied. I also find the text colouring a lot more readable here:
![image](https://user-images.githubusercontent.com/11966090/204844449-f3669a81-e411-4bed-a916-affc43acfd7b.png)

I've reverted to the previous version of the theme with qBittorent v4.5.0 as even with the bugs I find the visual style more readable and compact (possibly a Windows specific thing, your screen shots look a lot more consistent and compact). Happy to continue with suggestions or even collaborate if I can figure out the stylesheets and don't need to buy Qt dev tools.

**Comment by IDUnavailable** [comment]
I agree the other comments here as well as the two referenced issues. I like some of the changes but it definitely has some issues, although I don't think it needs *that* much tweaking to be great. IMO it's about readability, colors, and information density.

(I'll also note I'm on Firefox 112.0 for Win10x64, since I'm not sure what parts may be OS-dependent.)

---

![borderless](https://user-images.githubusercontent.com/8204892/232387929-5566b9fb-5927-4228-bc75-a27d48c2e937.png)

A lot of things need some clearer borders (or really any borders). The search tab is probably the one that needs it the most, since there's multiple elements that are just floating words in a void where you can't see what parts are **clickable**.

![old tabs](https://user-images.githubusercontent.com/8204892/232384421-90f37657-b671-4115-932b-d50ca222ec5f.png)

I don't know that I like the inactive tabs on the search page being light gray? Looks very odd and also isn't immediately readable to me when there's only two tabs (obviously when there's more your brain can quickly just go "the active one is the one that looks different"). I think they'd probably look better with subtle line borders instead of just using different background colors, but maybe that's not possible.

![new tabs](https://user-images.githubusercontent.com/8204892/232402479-98db73de-1e25-4e1f-9d72-a6ad1155b1de.png)

Again, not sure what's possible, and I don't know that this quick mockup I made is amazing either but it sort of illustrates what I'm picturing instead of "adjacent rectangles with light gray backgrounds".

---

![saturated](https://user-images.githubusercontent.com/8204892/232384518-547cad46-75b1-460e-b4ba-35994fd9e7b3.png)

I feel like these colors (and the ones in the status sidebar) need to be more pastel/desaturated to match the normal Dracula color scheme. More like this:

![desautrated](https://user-images.githubusercontent.com/8204892/232385270-9f3665b9-7a3b-4900-802e-c6bcd5832226.png)

I don't know exactly what I'm doing with regards to color but I the current ones just seem way too vibrant for a Dracula theme? (Ignore the fact that I lazily desaturated the background as well.)

---

![checkboxes](https://user-images.githubusercontent.com/8204892/232383820-3d5eb00e-2acc-4684-9c50-53088151b31d.png)

The checkboxs also look at-odds with the rest of the theme, but it also looks like some windows default so I don't know what control you have over them. Whenever I'm deleting a torrent the bright contrasting white box catches my eye and makes me think it might be checked.

![dracula for obsidian](https://user-images.githubusercontent.com/8204892/232404488-8b902427-d4bc-408f-a472-337029402cca.png)

Maybe something similar to the Dracula theme for Obsidian, if possible?

---

The only other thing is the spacing between rows of statuses/categories/tags/active torrents/search results seems a bit too large IMO.

That's all I can think of. Still the best theme in its current state though, so thanks for the work!

**Comment by MahdiMirzadeh** [comment]
it's been a few years,
But I've completely re-worked the styles template for both the qt-client and the webui implementation, working on qbittorrent v5.x and later

Also supporting a new set of themes as planned: dark, dracula, gruvbox-dark, solarized-dark, gruvbox-light, solarized-light

Read more at: https://github.com/MahdiMirzadeh/qbittorent

Closing the issue, because a lot of github issues have piled up over the years regarding the same stylesheet flaws (borders, non-visible scrollbar, etc.

Re-open if necessary

Kind Regards, Mahdi Mirzadeh
